component/topology/syncer.go

```diff
@@ -165,3 +189,29 @@ func isContextDone(ctx context.Context) bool {
 	}
 	return false
 }
+
+// ServerInfo is server static information.
+// It will not be updated when server running.
+// So please only put static information in ServerInfo struct.
+type ServerInfo struct {
+	GitHash        string `json:"git_hash"`
+	IP             string `json:"ip"`
+	Port           uint64 `json:"listening_port"`
+	StartTimestamp int64  `json:"start_timestamp"`
+}
+
+func getServerInfo() *ServerInfo {
+	info := &ServerInfo{
+		GitHash:        printer.NGMGitHash,
+		IP:             "",
+		Port:           0,
+		StartTimestamp: time.Now().Unix(),
+	}
+	cfg := config.GetGlobalConfig()
+	host, port, err := net.SplitHostPort(cfg.AdvertiseAddress)
+	if err == nil {
+		info.IP = host
+		info.Port, _ = strconv.ParseUint(port, 10, 64)
+	}
```

Comment:
What if there are errors?

Reply:
Already add test for it. If error, the port will be 0

Reply:
I don't think it is a good idea to spread invalid data to more places. 0 is definitely invalid data that will cause other components to broken, e.g. tidb-dashboard will definitely not sending data to the ngm as port 0 is invalid.